code/datums/quirks/negative_quirks/junkie.dm

```diff
@@ -19,11 +18,21 @@
 	var/next_process = 0 //! ticker for processing
 	var/drug_flavour_text = "Better hope you don't run out..."
 
+/datum/quirk_constant_data/junkie
+	associated_typepath = /datum/quirk/item_quirk/junkie
+	customization_options = list(/datum/preference/choiced/junkie)
+
 /datum/quirk/item_quirk/junkie/add_unique(client/client_source)
 	var/mob/living/carbon/human/human_holder = quirk_holder
 
+	// Avoid setting reagent_type for the Smoker and Alcoholic subtypes.
+	if(type == /datum/quirk/item_quirk/junkie)
```

Comment:
This should be refactored into an Addict path with subtypes for drugs, smokes, and booze

Reply:
Do this please. it's silly to use inheritance this way.